backend/server/floorplanets_server.go

```diff
@@ -1,21 +1,57 @@
-// Package server encapsulates operations on the http server
+// Package server contains the core of the flooplanets server
 package server
 
+import (
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+
+	"github.com/lanets/floorplanets/backend/app"
+	"github.com/lanets/floorplanets/backend/models"
+	"github.com/lanets/floorplanets/backend/server/internal/http"
+)
+
 type FloorplanetsServer struct {
-	httpServer *HttpServer
+	httpServer *http.HttpServer
+	database   *gorm.DB
+	app        *app.App
 }
 
 func NewFloorplanetsServer(address string) (*FloorplanetsServer, error) {
 
-	httpServer, err := NewHttpServer(address)
+	database, err := setupDatabase()
+	if err != nil {
+		return nil, err
+	}
+
+	application := app.NewApp(database)
+
+	httpServer, err := http.NewHttpServer(application, address)
+	if err != nil {
+		return nil, err
+	}
+
+	floorplanetsServer := FloorplanetsServer{
+		httpServer: httpServer,
+		database:   database,
+		app:        application,
+	}
+
+	return &floorplanetsServer, nil
+}
 
+func setupDatabase() (*gorm.DB, error) {
+	database, err := gorm.Open("sqlite3", "database.sqlite")
```

Comment:
Yeah well eventually it won't even be sqlite  and we will need to read host/port/username from somewhere.